### ISSUE TITLE
Cannot Interrupt Surgery when performing Cavity Implant

### ISSUE THREAD

**Issue by Penguaro** [tgstation/tgstation#27956]
When attempting to perform a Cavity Implant procedure on Chest, attempting to interrupt the surgery at the time when the player would implant something, places the surgical drapes in the patient instead of removing the drapes. Cavity was empty, Cautery was in other hand. Not sure if it is significant, we had just unsuccessfully tried to place an implant there that didn't fit (which necessitated ending the surgery).

**Comment by octareenroon91** [comment]
Cavity implant is kind of indiscriminate.

**Comment by XDTM** [comment]
probably a ..() has to go before the block of code instead of after somewhere

**Comment by octareenroon91** [comment]
https://github.com/tgstation/tgstation/blob/master/code/modules/surgery/cavity_implant.dm

````
//handle cavity
/datum/surgery_step/handle_cavity
	name = "implant item"
	accept_hand = 1
	accept_any_item = 1
	time = 32
	var/obj/item/IC = null
````

I don't think it's a stretch to assume accept_any_item means what it sounds like. More to the point, there's not much of a check on what's allowed to implant.

**Comment by octareenroon91** [comment]
It would be fairly straightforward to add a pop-up to cancel the step if using a surgical tool or bedsheet, so at least you don't lose tools into the cavity.

I am a little confused about the sheet not triggering a cancellation attempt.

**Comment by Penguaro** [comment]
To give a little more detail, we tried to just finish the surgery with the cautery and that attempted to insert the cautery into the cavity as well. The only way to finish was to leave the drapes in and then cautery. Then we grabbed a different set of drapes, cavity implant surgery and removed the drapes. After removing the inserted drapes as part of a new surgery we were able to cautery to end it. It seems like if you don't remove something new or implant something new, it won't let you do anything else.
